### DISCUSSION TITLE
Arduino I2C (Wire) Issues?

### DISCUSSION THREAD

**Discussion by mrengineer7777** [espressif/arduino-esp32#6419] [Q&A] [answered]
I recall at some point in the past reading about issues using I2C with arduino-esp32 as a master or maybe it was slave.  Does anyone else remember this and can you tell me if it was solved?
https://github.com/espressif/arduino-esp32/tree/master/libraries/Wire

Looking around, it seems there have been some fixes:
https://github.com/espressif/arduino-esp32/pull/5683 (replaces 5664)  Thread-Safe I2C based on IDF
https://github.com/espressif/arduino-esp32/pull/5746 Added I2C Slave (this is likely what I was thinking about)
https://github.com/espressif/arduino-esp32/pull/5770 (documentation)
https://github.com/espressif/arduino-esp32/pull/5910 Fixes clock stretching issue
https://github.com/espressif/arduino-esp32/pull/5935 Match vartype to IDF

Possible issues:
https://github.com/espressif/arduino-esp32/issues/6333  I2C scan. Possible user error.
https://github.com/espressif/arduino-esp32/issues/5934  (Duplicate 5972)  I2C Master Wire.requestFrom timeout not obeyed
https://github.com/espressif/arduino-esp32/issues/5907 I2C slave fires onRequest even though master was requesting a different address

FYI I'm currently using a custom build which has been fantastic:
platform = https://github.com/tasmota/platform-espressif32/releases/download/v2.0.2.3/platform-espressif32-2.0.2.3.zip

**Comment by Jason2866** [comment, marked as answer]
I dont know if i2c slave is issue free. We do not use in Tasmota. For i2c (master) i can say it just works :-)
